universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -0,0 +1,109 @@
+package controllers.admin;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import forms.QuestionForm;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import play.data.Form;
+import play.data.FormFactory;
+import play.mvc.Controller;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import services.question.InvalidPathException;
+import services.question.QuestionDefinition;
+import services.question.QuestionService;
+import services.question.UnsupportedQuestionTypeException;
+import views.admin.questions.QuestionEditView;
+import views.admin.questions.QuestionsListView;
+
+public class QuestionController extends Controller {
+
+  private final QuestionService service;
+  private final QuestionsListView listView;
+  private final QuestionEditView editView;
+  private final FormFactory formFactory;
+
+  @Inject
+  public QuestionController(
+      QuestionService service,
+      QuestionsListView listView,
+      QuestionEditView editView,
+      FormFactory formFactory) {
+    this.service = checkNotNull(service);
+    this.listView = checkNotNull(listView);
+    this.editView = checkNotNull(editView);
+    this.formFactory = checkNotNull(formFactory);
+  }
+
+  public CompletionStage<Result> list(String renderAs) {
+    return service
+        .getReadOnlyQuestionService()
+        .thenApplyAsync(
+            readOnlyService -> {
+              return ok(listView.render(readOnlyService.getAllQuestions(), renderAs));
+            });
+  }
+
+  public Result create(Request request) {
```

Comment:
suggestion (blocking): This should be `edit`: https://github.com/seattle-uat/universal-application-tool#routing-and-controller-methods